RecoTauTag/Configuration/python/tools/adaptToRunAtMiniAOD.py

```diff
@@ -7,241 +7,293 @@
+		self.miniAODTausTask.remove(getattr(self.process,'pfRecoTauTagInfoProducer'+self.postfix))
+
+		self.miniAODTausTask.remove(getattr(self.process,'recoTauAK4PFJets08Region'+self.postfix))
+		setattr(self.process,'recoTauAK4Jets08RegionPAT'+self.postfix,
+			cms.EDProducer("RecoTauPatJetRegionProducer",
+				       deltaR = self.process.recoTauAK4PFJets08Region.deltaR,
+				       maxJetAbsEta = self.process.recoTauAK4PFJets08Region.maxJetAbsEta,
+				       minJetPt = self.process.recoTauAK4PFJets08Region.minJetPt,
+				       pfCandAssocMapSrc = cms.InputTag(""),
+				       pfCandSrc = cms.InputTag("packedPFCandidates"),
+				       src = cms.InputTag(jetCollection)
+		))
+		self.miniAODTausTask.add(getattr(self.process,'recoTauAK4Jets08RegionPAT'+self.postfix))
+		if self.runBoosted:
+			getattr(self.process,'recoTauAK4Jets08RegionPAT'+self.postfix).pfCandAssocMapSrc = cms.InputTag('boostedTauSeedsPAT'+self.postfix, 'pfCandAssocMapForIsolation')
+			getattr(self.process,'recoTauPileUpVertices'+self.postfix).src = "offlineSlimmedPrimaryVertices"
+
+		for moduleName in self.miniAODTausTask.moduleNames():
+			self.convertModuleToMiniAODInput(moduleName)
+
+
+		# Adapt TauPiZeros producer
+		getattr(self.process,'ak4PFJetsLegacyHPSPiZeros'+self.postfix).builders[0].qualityCuts.primaryVertexSrc = "offlineSlimmedPrimaryVertices"
+		getattr(self.process,'ak4PFJetsLegacyHPSPiZeros'+self.postfix).jetSrc = jetCollection
+
+		# Adapt TauChargedHadrons producer
+		for builder in getattr(self.process,'ak4PFJetsRecoTauChargedHadrons'+self.postfix).builders:
+			builder.qualityCuts.primaryVertexSrc = "offlineSlimmedPrimaryVertices"
+			if builder.name.value() == 'tracks': #replace plugin based on generalTracks by one based on lostTracks
+				builder.name = 'lostTracks'
+				builder.plugin = 'PFRecoTauChargedHadronFromLostTrackPlugin'
+				builder.srcTracks = "lostTracks"
+				if self.runBoosted:
+					builder.dRcone = 0.3
+					builder.dRconeLimitedToJetArea = True
+		getattr(self.process,'ak4PFJetsRecoTauChargedHadrons'+self.postfix).jetSrc = jetCollection
+
+		# Adapt combinatoricRecoTau producer
+		getattr(self.process,'combinatoricRecoTaus'+self.postfix).jetRegionSrc = 'recoTauAK4Jets08RegionPAT'+self.postfix
+		getattr(self.process,'combinatoricRecoTaus'+self.postfix).jetSrc = jetCollection
+		# Adapt builders
+		for builder in getattr(self.process,'combinatoricRecoTaus'+self.postfix).builders:
+			for name,value in builder.parameters_().items():
+				if name == 'qualityCuts':
+					builder.qualityCuts.primaryVertexSrc = 'offlineSlimmedPrimaryVertices'
+				elif name == 'pfCandSrc':
+					builder.pfCandSrc = 'packedPFCandidates'
+		# Adapt supported modifiers and remove unsupported ones
+		modifiersToRemove_ = cms.VPSet()
+		for mod in getattr(self.process,'combinatoricRecoTaus'+self.postfix).modifiers:
+			if mod.name.value() == 'elec_rej':
+				modifiersToRemove_.append(mod)
+				continue
```

Comment:
this part, to L168 is particularly heavy on `getattr(self.process,'combinatoricRecoTaus'+self.postfix)`, which apparently points to the same object in this context/range of lines (the worst is the last one where `getattr` is called in a loop.
Please replace with access to an explicit object.

Reply:
Done